magenta/lib/melodies_lib.py

```diff
@@ -248,6 +254,10 @@ def _get_last_on_off_events(self):
         return (i, last_off)
     raise ValueError('No events in the stream')
 
+  def copy(self):
+    """Returns a deep copy of this MonphonicMelody object."""
```

Comment:
nit: probably shouldn't call this a "deep" copy because it uses the shallow copy operator (__copy__). I think in practice it does end up being a deep copy, but that's only because from_event_list happens to use an iterator over a list of ints.